### PR TITLE
refactor: move providers to separate folder

### PR DESCRIPTION
**Pull request by j3lte**
- Proposed changes for #99, moved all providers to their own dedicated file.
- Testing is still done in the main `create_oauth2_client.ts` file

### CODE REVIEW

**Comment by iuioiua** [comment]
Thanks for this, @j3lte! As this is an internal change, the main question is: does this simplify future maintenance? To me, the answer isn't super-obvious. I shared my thoughts at https://github.com/denoland/deno_kv_oauth/issues/99#issuecomment-1595883265.

I'm keen to hear some thoughts.

CC @kt3k, @lino-levan

**Comment by j3lte** [comment]
@iuioiua I did see your question in #99. To me it does simplify maintenance, because:

- If I'd need to change something to one of the providers, I can simply open the separate file and change it there.
- If I want to add a new one, I can take one of the providers and copy/paste them, changing things around.
- In my own experience, smaller files means better maintenance and faster iterations...

It's just a proposal, it's up to you guys if you'd like to use it. I'm basically taking the structure out of the [Next Auth playbook](https://github.com/nextauthjs/next-auth/tree/main/packages/core/src/providers)

File: src/create_oauth2_client.ts [hunk 1]
```diff
@@ -1,360 +1,16 @@
...
-  return new OAuth2Client({
-    clientId: Deno.env.get("TWITTER_CLIENT_ID")!,
-    clientSecret: Deno.env.get("TWITTER_CLIENT_SECRET")!,
-    authorizationEndpointUri: "https://twitter.com/i/oauth2/authorize",
-    tokenUri: "https://api.twitter.com/2/oauth2/token",
-    ...additionalOAuth2ClientConfig,
-  });
-}
+export type WithScope = { defaults: { scope: string | string[] } };
+export type WithRedirectUri = { redirectUri: string };
```
Comment (iuioiua) [inline src/create_oauth2_client.ts:5]: Maybe move to `src/providers/types.ts`?

File: src/create_oauth2_client.ts [hunk 2]
```diff
@@ -1,360 +1,16 @@
...
+
+export { createDiscordOAuth2Client } from "./providers/discord.ts";
```
Comment (iuioiua) [inline src/create_oauth2_client.ts:7]:
```suggestion
export * from "./providers/discord.ts";
```
I think it's fine to do this for all of these.

File: src/create_oauth2_client.ts (hunk 1 above)
Comment (j3lte) [reply src/create_oauth2_client.ts:5]: I'll move them to `src/providers/_types.ts`, makes more sense as it isn't a provider itself

**Comment by lino-levan** [comment]
This seems reasonable. I'd consider it a nice cleanup.

File: src/create_oauth2_client.ts
Comment (iuioiua) [inline src/create_oauth2_client.ts:1]: Let's rename this to `providers_test.ts`.

**Comment by kt3k** [comment]
Looks a reasonable change to me too

**Comment by iuioiua** [comment]
Alright then, let's go ahead with this 👍🏾

@j3lte, could you please address my recent suggestion and rebase?

**Comment by j3lte** [comment]
Merged in the main branch and moved `Spotify` also to a separate provider

**Pull request merged**